tests/src/ilasm/System/Runtime/CompilerServices/MethodImplOptionsTests.cs

```diff
@@ -0,0 +1,152 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Whether in this PR or a later one, it would be good to add a test that confirms a method marked with the AggressiveOptimization attribute doesn't JIT twice. Now that EventPipe appears to work uniformly across platforms and RuntimeEventSource is implemented I think you could build a test that solely uses an in-proc EventListener to observe the jit events. I should have a little sample laying around if its helpful though I haven't explored jit events specifically.

Reply:
I'll do this in a separate PR, I'll need to add a perf test as well. Filed https://github.com/dotnet/coreclr/issues/20229 to track both.